test/objspace/test_objspace.rb

```diff
@@ -164,6 +164,11 @@ def test_reachable_objects_size
     end;
   end
 
+  def test_trace_object_allocations_stop_first
```

Comment:
@nobu I added this test, but as I said I highly doubt it would catch a regression on CI. I guess it still has a documentation value.

Reply:
Using a subprocess would be a way to guarantee no `trace_object_allocations_start` was used before, right?

Reply:
Yes. @XrXr told me about `assert_separately` last night. I just updated the test to use it.

Reply:
You can use `assert_ruby_status` too.
For this issue, `assert_separately` reports an error while `assert_ruby_status` a failure.

Reply:
I'm not sure which is best. Do you want me to update the test to use `assert_ruby_status`?

Since we're just testing that it doesn't crash, I suppose reporting an error is fine?